poseidon/src/permutation.rs

```diff
@@ -42,6 +42,23 @@ pub fn full_round<F: Field, SC: SpongeConstants>(
     }
 }
 
+/// Same as full round but works better with arrays.
+pub fn full_round2<F: Field, SC: SpongeConstants>(

```

Comment:
Is it really useful?
Also, the same is not very concrete. I would get rid of this if not strictly required.

Reply:
See https://github.com/o1-labs/proof-systems/pull/1720